site/content/blog/2024-05-21-introducing-the-grpc-addon/index.md

```diff
@@ -0,0 +1,98 @@
+---
+title: "Introducing the gRPC Add-on"
+summary: "Introducing a new add-on for viewing and attacking gRPC endpoints."
+images:
+- https://www.zaproxy.org/blog/2024-05-21-introducing-the-grpc-addon/images/0-blog-grpc-intro.png
+type: post
+tags:
+- blog
+- grpc
+date: "2024-05-21"
+authors:
+    - amit
+---
+[gRPC](https://grpc.io/) APIs have become increasingly popular for building efficient and robust microservices architectures. 
+However, testing these APIs effectively can be challenging, especially when dealing with Protobuf message encoding and decoding. 
+In this blog post, we'll explore a custom extension designed to streamline the process of testing gRPC endpoints and handling Protobuf messages.
+
+![ZAPbot attacking gRPC](images/0-blog-grpc-intro.png)
+
+The add-on is still in an early stage, so the range of its functionality is limited. 
+However, you can combine it with existing ZAP functionality to abuse gRPC endpoints in many different ways.
+
+This extension aims to simplify the testing process by providing tools for:
+
+* **Encoding and Decoding** : Easily encode and decode Protobuf messages for testing purposes.
+* **Testing gRPC Endpoints** : You can combine it with existing ZAP functionality to abuse gRPC endpoints in many different ways.
+
+### Installation
+
+If you are using the latest version of ZAP then you can download the gRPC add-on from the 
+[ZAP Marketplace](/addons/).
+
+### How to Display a gRPC Decoded Message
+
+![Displaying a gRPC decoded message](images/gRPCResponseBody.png)
```

Comment:
Before it could be added this sentence: https://github.com/zaproxy/zap-extensions/blob/d8bec89480f24d044110a607b9ee4f6cbd462f85/addOns/grpc/src/main/javahelp/org/zaproxy/addon/grpc/resources/help/contents/grpc.html#L22